### PR TITLE
[OF-1027] feat!: Update ResetUserPassword to token-change-password endpoint

### PR DESCRIPTION
**Pull request by Mag-JB**
connected-spaces-platform Pull Request

**For the reviewer**

* [ ] If required, are the changes covered by appropriate tests?
* [ ] Are any public-facing API changes well documented?
* [ ] Is the code easily readable and extensible and/or follow existing conventions?

### CODE REVIEW

**Comment by Mag-JB** [comment]
This has been manually tested as integration tests are not possible

**Comment by MAG-SamBirley** [comment]
@MAG-DanielMurdolo with this change, we're removing the parameter that lets client code specify the redirect URL associated with a reset password invocation. Wanted to check that is expected from your side?

**Review by MAG-SamBirley — DISMISSED** [review]
Great work, thanks!

**Comment by MAG-AdrianMeredith** [comment]
I can confirm that the Web Client doesn't currently use this function and as I understand it, not redirect url should be necessary.  Should be good our side

File: Library/include/CSP/Systems/Users/UserSystem.h [hunk 1]
```diff
@@ -162,8 +162,10 @@ class CSP_API CSP_NO_DISPOSE UserSystem : public SystemBase
 	CSP_ASYNC_RESULT void ConfirmUserEmail(NullResultCallback Callback);
 
 	/// @brief Reset the users password.
+	/// @param Token csp::common::String : Bearer Token of the user changing their password
```
Comment (MAG-pw) [inline Library/include/CSP/Systems/Users/UserSystem.h:165]: I think of bearer token as the OAuth access token in the headers. Is that what this token is or is it something else?

**Review by MAG-pw — COMMENTED** [review]
Looks generally ok, just one quick question on the comments.

File: Library/include/CSP/Systems/Users/UserSystem.h (hunk 1 above)
Comment (MAG-SamBirley) [reply Library/include/CSP/Systems/Users/UserSystem.h:165]: Good catch. According to swagger this is actually a token that is received via an automated email that is sent

Comment (MAG-pw) [reply Library/include/CSP/Systems/Users/UserSystem.h:165]: I thought that might be what this was. I think we should update the comment to make that more clear before shipping this.

Comment (Mag-JB) [reply Library/include/CSP/Systems/Users/UserSystem.h]: sorted

**Review by MAG-SamBirley — APPROVED** [review]

**Review by MAG-pw — APPROVED** [review]

**Pull request merged**